src/main/java/controller/BlackJackController.java

```diff
@@ -39,50 +45,98 @@ private List<String> initPlayerNames() {
+        }
+    }
+
+    private void initDistributeCards(BlackJackGame blackJackGame) {
+        List<String> playerNames = blackJackGame.getPlayerNames();
         ResultView.printInitMessage(playerNames);
-        ResultView.printParticipantResult(Constants.DEALER_NAME, blackJackGame.findCardNamesByParticipantName(Constants.DEALER_NAME));
+        ParticipantResultDto dealerResultDto = blackJackGame.generateParticipantResultByParticipantName(Constants.DEALER_NAME);
+        ResultView.printParticipantResult(dealerResultDto);
         for (String playerName : playerNames) {
-            List<String> findCardNames = blackJackGame.findCardNamesByParticipantName(playerName);
-            ResultView.printParticipantResult(playerName, findCardNames);
+            ParticipantResultDto playerResultDto = blackJackGame.generateParticipantResultByParticipantName(playerName);
+            ResultView.printParticipantResult(playerResultDto);
+        }
+    }
+
+    private void additionalCardDrawStep(BlackJackGame blackJackGame) {
+        if (!isDealerBlackjack(blackJackGame)) {
+            additionalCardDraw(blackJackGame);
+        }
+    }
+
+    private boolean isDealerBlackjack(BlackJackGame blackJackGame) {
+        return blackJackGame.isDealerBlackjack();
+    }
+
+    private void additionalCardDraw(BlackJackGame blackJackGame) {
+        askPlayerAdditionalCardDrawStep(blackJackGame);
+        dealerAdditionalCardDrawStep(blackJackGame);
+    }
+
+    private BettingMoney generateBettingMoney(String playerName) {
+        try {
+            OutputView.printInputBettingMoney(playerName);
+            String bettingMoney = InputView.inputBettingMoney();
+            return new BettingMoney(bettingMoney);
+        } catch (IllegalArgumentException e) {
+            OutputView.printMessage(e.getMessage());
+            return generateBettingMoney(playerName);
         }
     }
 
-    public void playerDrawCardStep(BlackJackGame blackJackGame) {
-        Players players = blackJackGame.getPlayers();
-        for (Player player : players.getPlayers()) {
-            drawCardOrPass(blackJackGame, player);
+    private void askPlayerAdditionalCardDrawStep(BlackJackGame blackJackGame) {
+        List<String> playerNames = blackJackGame.getPlayerNames();
+        for (String playerName : playerNames) {
+            askPlayerAdditionalCardDraw(blackJackGame, playerName);
         }
     }
 
-    private void drawCardOrPass(BlackJackGame blackJackGame, Player player) {
+    private void askPlayerAdditionalCardDraw(BlackJackGame blackJackGame, String playerName) {
         AdditionalDrawStatus additionalDrawStatus = AdditionalDrawStatus.DRAW;
-        while (AdditionalDrawStatus.isDrawable(additionalDrawStatus) && blackJackGame.canPlayerDrawCard(player)) {
-            String playerName = player.getName();
+        while (AdditionalDrawStatus.isDrawable(additionalDrawStatus) && blackJackGame.canPlayerDrawCard(playerName)) {
```

Comment:
additionalDrawStatus에게 직접 물어보는게 어떨까요?

```suggestion
        while (additionalDrawStatus.isDrawable() && blackJackGame.canPlayerDrawCard(playerName)) {
```